actionview/lib/action_view/base.rb

```diff
@@ -197,17 +198,49 @@ def assign(new_assigns) # :nodoc:
       @_assigns = new_assigns.each { |key, value| instance_variable_set("@#{key}", value) }
     end
 
-    def initialize(context = nil, assigns = {}, controller = nil, formats = nil) #:nodoc:
+    # :stopdoc:
+
+    def self.build_renderer(context, controller, formats)
+      lookup_context = context.is_a?(ActionView::LookupContext) ?
+        context : ActionView::LookupContext.new(context)
+      lookup_context.formats  = formats if formats
+      lookup_context.prefixes = controller._prefixes if controller
+      ActionView::Renderer.new(lookup_context)
+    end
+
+    def self.empty
+      with_view_paths([])
+    end
+
+    def self.with_view_paths(view_paths, assigns = {}, controller = nil)
+      with_context ActionView::LookupContext.new(view_paths), assigns, controller
+    end
+
+    def self.with_context(context, assigns = {}, controller = nil)
+      new ActionView::Renderer.new(context), assigns, controller
+    end
+
+    NULL = Object.new
+
+    # :startdoc:
+
+    def initialize(context, assigns = {}, controller = nil, formats = NULL) #:nodoc:
```

Comment:
What do you think about renaming the `context` argument to `renderer` or similar?

Reply:
Ya, I think we should.